happi/cli.py

```diff
@@ -97,223 +47,314 @@ def happi_cli(args):
-        if is_invalid_field:
-            sys.exit(1)
+            client_args[criteria] = fnmatch.translate(value)
+
+    regex_list = client.search_regex(**client_args)
+    results = regex_list + range_list
+
+    # find the repeated items
+    res_size = len(results)
+    repeated = []
+    for i in range(res_size):
+        k = i + 1
+        for j in range(k, res_size):
+            if results[i] == results[j] and results[i] not in repeated:
+                repeated.append(results[i])
+
+    # we only want to return the ones that have been repeated when
+    # they have been matched with both search_regex() & search_range()
+    if repeated:
+        final_results = repeated
+    elif regex_list and not is_range:
+        # only matched with search_regex()
+        final_results = regex_list
+    elif range_list and is_range:
+        # only matched with search_range()
+        final_results = range_list
+    else:
+        final_results = []
+
+    if json:
+        json.dump([dict(res.item) for res in final_results], indent=2,
+                  fp=sys.stdout)
+    elif names:
+        out = " ".join([res.item.name for res in final_results])
+        click.echo(out)
+    else:
+        for res in final_results:
+            res.item.show_info()
+
+    if not final_results:
+        logger.error('No devices found')
+    return final_results
+
+
+@happi_cli.command()
+@click.option('--clone', default='',
+              help='Copy the fields from an existing container. '
+              'Provide the name of the item to clone.')
+@click.pass_context
+def add(ctx, clone):
+    """Add new entries interactively."""
+    logger.debug(f'Starting interactive add, {clone}')
+    # retrieve client
+    client = ctx.obj
+
+    registry = happi.containers.registry
+    if clone:
+        clone_source = client.find_device(name=clone)
+        # Must use the same container if cloning
+        response = registry.entry_for_class(clone_source.__class__)
+    else:
+        clone_source = None
+        # Keep Device at registry for backwards compatibility but filter
+        # it out of new devices options
+        options = os.linesep.join(
+            [k for k, _ in registry.items() if k != "Device"]
+        )
+        ctnr_prompt = (
+            'Please select a container, or press enter for generic '
+            f'Ophyd Device container: {os.linesep}{options}'
+            '{}{}Selection'.format(os.linesep, os.linesep)
+        )
+        logger.debug(ctnr_prompt)
+        response = click.prompt(ctnr_prompt, default='OphydItem')
+        if response and response not in registry:
+            logger.info(f'Invalid device container {response}')
+            return
+
+    container = registry[response]
+    logger.debug(f'Contaner selected: {container.__name__}')
+
+    # Collect values for each field
+    kwargs = {}
+    for info in container.entry_info:
+        item_value = prompt_for_entry(info, clone_source=clone_source)
```

Comment:
I entered an invalid value for a `float` and `happi add` exited on me:
```
Enter value for z, enforce=float [-1.0]: z
...
  File "/Users/klauer/Repos/happi/happi/prompt.py", line 76, in prompt_for_entry
    value = click.prompt(val_prompt, default=default,
  File "/Users/klauer/mc/envs/lucid/lib/python3.9/site-packages/click/termui.py", line 175, in prompt
    result = value_proc(value)
ValueError: could not convert string to float: 'z'
(lucid) PC98125:happi klauer$

```

Reply:
Slight differences in what kinds of exceptions `click.prompt` catches lead to this.  `UsageError` is the only exception that can be raised and continue prompting.  I had our `EnforceError` subclass `UsageError`, but I didn't fully cover our enforce_value function in my previous enforce_doc work.  

This also led to fixes in the regex handling area. 👍

Reply:
I need to add tests for this in general, maybe tests also go in a new PR.  (I have a bunch of stashed changes fixing deprecation warnings for happi.device.Device)